proxy/src/main/java/net/md_5/bungee/UserConnection.java

```diff
@@ -339,8 +339,7 @@ public void disconnect0(final BaseComponent... reason)
                 @Override
                 public void run()
                 {
-                    unsafe().sendPacket( new Kick( ComponentSerializer.toString( reason ) ) );
-                    ch.close();
+                    ch.close( new Kick( ComponentSerializer.toString( reason ) ) );
                 }
             }, 500, TimeUnit.MILLISECONDS );
```

Comment:
Why didn't you remove the delay like in https://github.com/SpigotMC/BungeeCord/pull/1641 ?